webapp/src/ts/services/transitions/muting.transition.ts

```diff
@@ -0,0 +1,442 @@
+  private readonly MUTE_PROPERTY = 'mute_forms';
+  private readonly UNMUTE_PROPERTY = 'unmute_forms';
+  private readonly CLIENT = 'client';
+  private readonly SERVER = 'server';
+
+  private getConfig(settings = {}) {
+    return settings[this.CONFIG_NAME] || {};
+  }
+
+  private getMutingForms() {
+    return this.transitionConfig?.[this.MUTE_PROPERTY] || [];
+  }
+
+  getUnmutingForms(settings?) {
+    const config = settings ? this.getConfig(settings) : this.transitionConfig;
+    return config?.[this.UNMUTE_PROPERTY] || [];
+  }
+
+  init(settings) {
+    this.transitionConfig = this.getConfig(settings);
+
+    const mutingForms = this.getMutingForms();
+    if (!mutingForms || !Array.isArray(mutingForms) || !mutingForms.length) {
+      console.warn(
+        `Configuration error. Config must define have a '${this.CONFIG_NAME}.${this.MUTE_PROPERTY}' array defined.`
+      );
+      this.inited = false;
+    } else {
+      this.inited = true;
+    }
+
+    return this.inited;
+  }
+
+  private isMuteForm(form) {
+    return !!form && this.getMutingForms().includes(form);
+  }
+
+  isUnmuteForm(form, settings?) {
+    return !!form && this.getUnmutingForms(settings).includes(form);
+  }
+
+  /**
+   * Returns whether a document is a muting or unmuting report that should be processed.
+   * We only process new reports. The muting transition should not run when reports are edited.
+   * @param {Doc} doc - the doc to check
+   * @returns {Boolean} - whether this is a new muting or unmuting report
+   * @private
+   */
+  private isRelevantReport(doc) {
+    // exclude docs that are not reports and existent reports.
+    if (!doc || doc._rev || doc.type !== 'data_record' || !doc.form) {
+      return false;
+    }
+
+    if (this.isMuteForm(doc.form) || this.isUnmuteForm(doc.form)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Returns whether a document is a new contact.
+   * The muting transition should not run on when existing contacts are edited.
+   * @param {Object} doc - doc to check
+   * @returns {Boolean} - whether the doc is a new contact type document
+   * @private
+   */
+  private isRelevantContact(doc) {
+    return !doc._rev && this.contactTypesService.includes(doc);
+  }
+
+  /**
+   * @param {Array<Doc>} docs - docs to be saved
+   * @return {Boolean} - whether any of the docs from the batch should be processed
+   */
+  filter(docs) {
+    return !!docs.filter(doc => this.isRelevantReport(doc) || this.isRelevantContact(doc)).length;
+  }
+
+  private async hydrateDocs(context:MutingContext) {
+    const docs = [
+      ...context.reports,
+      ...context.contacts,
+    ];
+    const hydratedDocs = await this.lineageModelGeneratorService.docs(docs);
+
+    for (const doc of hydratedDocs) {
+      this.storeHydratedDoc(doc, context);
+      this.storeHydratedDoc(doc.patient, context);
+      this.storeHydratedDoc(doc.place, context);
+    }
+  }
+
+  private storeHydratedDoc(doc, context) {
```

Comment:
I feel a little uncomfortable with this being called `store...` - I keep thinking it's going to write it to the DB. What about `update...` or `cache...`?

Reply:
`cache` it is! Thanks!